src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private final int startPositionForSubstringDate = 0;
```

Comment:
Consider encapsulating the logic for parsing the date from a string into a separate method. This would avoid the need for these constants and make the code more readable.